### PR TITLE
Update Boto3 API calls in ECSOperator

### PR DESCRIPTION
**Pull request by scottypate**
This PR corrects two errors that can occur when utilizing the parameter `reattach=True`. These two errors are detailed below. 

1. The `describe_task_definition()` function requires a named keyword argument instead of a positional argument. The current error generated is...

```
[2021-05-25 14:33:24,145] {{taskinstance.py:1455}} ERROR - describe_task_definition() only accepts keyword arguments.
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1112, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1285, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1310, in _execute_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/operators/ecs.py", line 193, in execute
    self._try_reattach_task()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/operators/ecs.py", line 244, in _try_reattach_task
    task_def_resp = self.client.describe_task_definition(self.task_definition)
  File "/home/airflow/.local/lib/python3.8/site-packages/botocore/client.py", line 354, in _api_call
    raise TypeError(
TypeError: describe_task_definition() only accepts keyword arguments.
```

2. The `list_tasks()` function cannot provide both `launch_type` and `family` as a filter criteria. The current error generated is...

```
[2021-05-25 12:32:18,441] {{taskinstance.py:1455}} ERROR - An error occurred (InvalidParameterException) when calling the ListTasks operation: cannot specify filters 'launchType' and 'family' at the same time
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1112, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1285, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1310, in _execute_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/operators/ecs.py", line 193, in execute
    self._try_reattach_task()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/operators/ecs.py", line 247, in _try_reattach_task
    list_tasks_resp = self.client.list_tasks(
  File "/home/airflow/.local/lib/python3.8/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/botocore/client.py", line 676, in _make_api_call
    raise error_class(parsed_response, operation_name)
```

When creating a task definition with the same family name but changing the `launch_type`, then it creates a new revision of the task with the same family name. I removed this `launch_type` parameter from this function. The ECS task will be running the latest version of the task definition and I don't think it would be possible to be running the same family of task definition on multiple `launch_type` strategies.

### CODE REVIEW

**Comment by scottypate** [comment]
👋 @darwinyip, since you added the `reattach` option to this operator I wonder if you would look at this PR and see if it makes sense to you.

**Review by ferruzzi — APPROVED** [review]
Seems like a straight_forward fix.

**Comment by potiuk** [comment]
Reopened to rebuild it.

**Comment by eladkal** [comment]
@scottypate can you rebase? it will most likely solve the CI build issue

File: airflow/providers/amazon/aws/operators/ecs.py [hunk 1]
```diff
@@ -266,11 +266,11 @@ def _start_task(self):
         self.arn = response['tasks'][0]['taskArn']
 
     def _try_reattach_task(self):
-        task_def_resp = self.client.describe_task_definition(self.task_definition)
+        task_def_resp = self.client.describe_task_definition(taskDefinition=self.task_definition)
         ecs_task_family = task_def_resp['taskDefinition']['family']
 
         list_tasks_resp = self.client.list_tasks(
-            cluster=self.cluster, launchType=self.launch_type, desiredStatus='RUNNING', family=ecs_task_family
+            cluster=self.cluster, desiredStatus='RUNNING', family=ecs_task_family
```
Comment (o-nikolas) [inline airflow/providers/amazon/aws/operators/ecs.py:273]: How did you decide which of the two arguments to drop?

Comment (RosterIn) [reply airflow/providers/amazon/aws/operators/ecs.py:273]: the op explained it in the PR description. I think it makes sense

Comment (o-nikolas) [reply airflow/providers/amazon/aws/operators/ecs.py:273]: Got it, must have missed that!

**Review by o-nikolas — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**